### DISCUSSION TITLE
How to add watermark, a half transparent overlay image

### DISCUSSION THREAD

**Discussion by suntong** [ImageMagick/ImageMagick#6594] [Help] [answered]
### ImageMagick version

6.9.11.60+dfsg-1.6

### Operating system, version and so on

Debian GNU/Linux 12 (bookworm)

### Description

Maybe it is as simple as using `-compose overlay`, but I'm looking for a script/way to add a grey image as a half transparent overlay to the SE corner of anther image.

How to do that with Imagemagick please?  
[This](https://github.com/ImageMagick/ImageMagick/discussions/5781#discussioncomment-4186660) might be the answer, but on my Debian GNU/Linux 12 (bookworm) machine, I don't have that command, as these are all I have installed:

```
/usr/bin/animate-im6.q16
/usr/bin/compare-im6.q16
/usr/bin/composite-im6.q16
/usr/bin/conjure-im6.q16
/usr/bin/convert-im6.q16
/usr/bin/display-im6.q16
/usr/bin/identify-im6.q16
/usr/bin/import-im6.q16
/usr/bin/mogrify-im6.q16
/usr/bin/montage-im6.q16
/usr/bin/stream-im6.q16
```

**Comment by snibgo** [comment]
Is ImageMagick installed on your computer?

What is the text output from the commands "`convert -version`" and "`magick -version`"?

**Reply by suntong** [reply]
```sh
$ apt-cache policy imagemagick
imagemagick:
  Installed: 8:6.9.11.60+dfsg-1.6
  Candidate: 8:6.9.11.60+dfsg-1.6
  Version table:
 *** 8:6.9.11.60+dfsg-1.6 500
        500 http://deb.debian.org/debian bookworm/main amd64 Packages
         80 http://deb.debian.org/debian trixie/main amd64 Packages
        100 /var/lib/dpkg/status

$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm

$ convert -version
Version: ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib

$ magick -version
bash: magick: command not found
```

**Comment by snibgo** [comment]
Okay, you have `convert`. So a simple command could be:
```
convert image.png wmark.png -compose over -composite out.png
```
Use your own filenames, of course. They can be full paths. They don't have to be png files; they could be tiff or jpg or gif or many others. For a list of formats you can use, type "`convert -list format`".

The watermark image `wmark.png` should have transparency.

Instead of `over`, you could use another method such as `overlay`. For a complete list, type "`convert -list compose`".

**Reply by suntong** [reply]
Thanks, I tried `convert -list compose` and saw that `Dissolve` is one of the options. Is it somehow possible to use convert + Dissolve to do what https://imagemagick.org/Usage/compose/#dissolve is doing? As the transparency can be controlled and the watermark image (eg https://imagemagick.org/Usage/images/star.gif) does not need to have transparency.

**Comment by snibgo** [comment, marked as answer]
Your IM is v6, which is quite old. The more recent IM is v7 and uses `magick`.

Where a documented command starts with "`magick composite`", you can use just "`composite`".

Where a documented command starts with just "`magick`" and is immediately followed by an image filename, you can use "`convert`" instead.
